core/service/team_test.go

```diff
@@ -0,0 +1,804 @@
+	}
+	team2 := entity.Team{
+		ID:            teamID2,
+		Name:          "TeamName2",
+		IconURL:       &iconURL2,
+		CreatorUserID: requesterUserID2,
+		OwnerUserID:   requesterUserID2,
+		CreatedAt:     now,
+		UpdatedAt:     &now,
+	}
+
+	// insert teams into table
+	if !assert.Nil(t, teamService.teamDaoV2.CreateTeam(ct, tx, team1)) {
+		return
+	}
+	if !assert.Nil(t, teamService.teamDaoV2.CreateTeam(ct, tx, team2)) {
+		return
+	}
+
+	teamFilter := TeamFilter{
+		TeamID: &teamID2,
+	}
+
+	teamsFound, internalErr := teamService.FindTeams(ct, &teamFilter)
+	if !assert.Nil(t, internalErr) {
+		return
+	}
+
+	// only team2 should be returned
+	assert.Equal(t, len(teamsFound), 1)
+	assert.Equal(t, team2.ID, teamsFound[0].ID)
+	assert.Equal(t, team2.Name, teamsFound[0].Name)
+	assert.Equal(t, team2.IconURL, teamsFound[0].IconURL)
+	assert.Equal(t, team2.CreatorUserID, teamsFound[0].CreatorUserID)
+	assert.Equal(t, team2.OwnerUserID, teamsFound[0].OwnerUserID)
+	assert.NotNil(t, team2.CreatedAt, teamsFound[0].CreatedAt)
+	assert.NotNil(t, team2.UpdatedAt, teamsFound[0].UpdatedAt)
+}
+
+func TestTeamService_FindTeamsForUser(t *testing.T) {
+	teamService, ok := prepareTeamService(t)
+	if !ok {
+		return
+	}
+
+	var requesterUserID1 uint64 = 20
+	var requesterUserID2 uint64 = 32
+	var teamID1 uint64 = 12
+	var teamID2 uint64 = 2
+	var teamID3 uint64 = 4
+	ct := context.Background()
+	ct = ctx.NewContextWithUserID(ct, requesterUserID1)
+
+	tx, err := teamService.transactionFactory.BeginTx(ct, nil)
+	if !assert.Nil(t, err) {
+		return
+	}
+	defer tx.Rollback()
+
+	var iconURL1 = "https://test1"
+	var iconURL2 = "https://test2"
+	var iconURL3 = "https://test3"
+	now := time.Now().UTC()
+	team1 := entity.Team{
+		ID:            teamID1,
+		Name:          "TeamName1",
+		IconURL:       &iconURL1,
+		CreatorUserID: requesterUserID1,
+		OwnerUserID:   requesterUserID1,
+		CreatedAt:     now,
+		UpdatedAt:     &now,
+	}
+	team2 := entity.Team{
+		ID:            teamID2,
+		Name:          "TeamName2",
+		IconURL:       &iconURL2,
+		CreatorUserID: requesterUserID1,
+		OwnerUserID:   requesterUserID1,
+		CreatedAt:     now,
+		UpdatedAt:     &now,
+	}
+	team3 := entity.Team{
+		ID:            teamID3,
+		Name:          "TeamName3",
+		IconURL:       &iconURL3,
+		CreatorUserID: requesterUserID1,
+		OwnerUserID:   requesterUserID1,
+		CreatedAt:     now,
+		UpdatedAt:     &now,
+	}
+
+	teamMember1 := entity.TeamMember{TeamID: teamID1, UserID: requesterUserID1, CreatedAt: now}
+	teamMember2 := entity.TeamMember{TeamID: teamID2, UserID: requesterUserID2, CreatedAt: now}
+	teamMember3 := entity.TeamMember{TeamID: teamID3, UserID: requesterUserID1, CreatedAt: now}
+
+	// insert teams and teamMembers into table
+	if !assert.Nil(t, teamService.teamDaoV2.CreateTeam(ct, tx, team1)) {
+		return
+	}
+	if !assert.Nil(t, teamService.teamDaoV2.CreateTeam(ct, tx, team2)) {
+		return
+	}
+	if !assert.Nil(t, teamService.teamDaoV2.CreateTeam(ct, tx, team3)) {
+		return
+	}
```

Comment:
```suggestion
	if !assert.Nil(t, teamService.teamDaoV2.CreateTeam(ct, tx, team1)) {
		return
	}
	
	if !assert.Nil(t, teamService.teamDaoV2.CreateTeam(ct, tx, team2)) {
		return
	}
	
	if !assert.Nil(t, teamService.teamDaoV2.CreateTeam(ct, tx, team3)) {
		return
	}
```
Let's return `Ref` struct in `prepareTeamService` and use `teamDaoV2` from `refs` instead to avoid breaking encapsulation.